### PR TITLE
Leon/lend app

### PR DESCRIPTION
**Pull request by leonkho**

### CODE REVIEW

File: lend/src/components/Balance/index.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+`;
+
+export const Balance = () => {
+  const { account } = useWeb3Wallet();
+  const Logo = getAssetLogo("USDC");
+  const { loading, accountBalances } = useVaultAccountBalances();
+  const yourBalance = accountBalances.totalBalance;
+  const rbnRewards = accountBalances.rbnEarned;
+  const yourDeposits = useVaultTotalDeposits();
+  const decimals = 6;
```
Comment (cjinghong) [inline lend/src/components/Balance/index.tsx:100]: can use `getAssetDecimals("USDC")` here

File: lend/src/components/Balance/index.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        parseFloat(formatUnits(yourDeposits, decimals))) *
+      100
+    );
+  }, [yourDeposits, yourBalance]);
+  const [triggerClaimModal, setClaimModal] = useState<boolean>(false);
+
+  return (
+    <>
+      <LendModal
+        show={Boolean(triggerClaimModal)}
```
Comment (cjinghong) [inline lend/src/components/Balance/index.tsx:120]: `triggerClaimModal` is already a boolean

File: lend/src/utils/blockies.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
+// The random number is a js implementation of the Xorshift PRNG
```
Comment (cjinghong) [inline lend/src/utils/blockies.js:1]: why not just import from `shared/` instead?

File: lend/src/utils/asset.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
+import React from "react";
```
Comment (cjinghong) [inline lend/src/utils/asset.ts:1]: this can also be imported from `shared/lib/utils/asset` @timongll

File: lend/src/utils/address.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const truncateAddress = (address: string) => {
+  return address.slice(0, 6) + "..." + address.slice(address.length - 4);
```
Comment (cjinghong) [inline lend/src/utils/address.ts]: @timongll `shared/lib/utils/address`

File: lend/src/utils/chainSwitching.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import { providers } from "ethers";
+
+// This error code indicates that
+enum ChainCodeErrorEnum {
```
Comment (cjinghong) [inline lend/src/utils/chainSwitching.ts:4]: @timongll `/shared/lib/utils`

File: lend/src/utils/math.ts [hunk 7]
```diff
@@ -0,0 +1,210 @@
+import { BigNumber, BigNumberish } from "@ethersproject/bignumber";
```
Comment (cjinghong) [inline lend/src/utils/math.ts:1]: this is also from `shared/lib/math`

File: webapp/tsconfig.json [hunk 8]
```diff
@@ -14,15 +17,17 @@
     "moduleResolution": "node",
     "resolveJsonModule": true,
     "isolatedModules": true,
-    "noEmit": false,
+    "noEmit": true,
```
Comment (cjinghong) [inline webapp/tsconfig.json:20]: this file should not be updated, it will affect `treasury`

**Review by cjinghong — CHANGES_REQUESTED** [review]

File: lend/src/utils/blockies.js (hunk 3 above)
Comment (timongll) [reply lend/src/utils/blockies.js]: removed some utils

File: lend/src/utils/asset.ts (hunk 4 above)
Comment (timongll) [reply lend/src/utils/asset.ts:1]: the `VaultOptions` and `Chains` used here is different

File: lend/src/utils/address.ts (hunk 5 above)
Comment (timongll) [reply lend/src/utils/address.ts]: removed

File: lend/src/utils/math.ts (hunk 7 above)
Comment (timongll) [reply lend/src/utils/math.ts:1]: `Chains` and `assets` different here

File: lend/src/utils/chainSwitching.ts (hunk 6 above)
Comment (timongll) [reply lend/src/utils/chainSwitching.ts]: removed

**Review by cjinghong — APPROVED** [review]

**Pull request merged**